flutter/test/mocks.dart

```diff
@@ -63,3 +63,54 @@ class MockPlatform implements Platform {
   @override
   bool get isFuchsia => (operatingSystem == 'fuchsia');
 }
+
+// Does nothing or throws.
+// Usefull for when a Hub needs to be passed but is not used.
+class NoOpHub implements Hub {
+  @override
+  void addBreadcrumb(Breadcrumb crumb, {hint}) {}
+
+  @override
+  void bindClient(SentryClient client) {}
+
+  @override
+  Future<SentryId> captureEvent(SentryEvent event,
+      {stackTrace, hint, ScopeCallback? withScope}) {
+    throw UnimplementedError();
```

Comment:
to make this a bit more useful then I'd return `SentryId.empty()` instead of `UnimplementedError` for all methods